src/main/kotlin/com/sourcegraph/cody/config/SourcegraphInstanceLoginDialog.kt

```diff
@@ -0,0 +1,248 @@
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.dsl.builder.MAX_LINE_LENGTH_NO_WRAP
+import com.intellij.ui.dsl.builder.panel
+import com.intellij.ui.dsl.gridLayout.HorizontalAlign
+import com.intellij.ui.layout.not
+import com.intellij.ui.layout.selected
+import com.intellij.util.ui.UIUtil.getInactiveTextColor
+import com.sourcegraph.cody.api.SourcegraphApiRequestExecutor
+import com.sourcegraph.cody.api.SourcegraphAuthenticationException
+import com.sourcegraph.cody.auth.SourcegraphAuthService
+import com.sourcegraph.cody.auth.SsoAuthMethod
+import com.sourcegraph.cody.config.DialogValidationUtils.custom
+import com.sourcegraph.cody.config.DialogValidationUtils.notBlank
+import com.sourcegraph.common.AuthorizationUtil
+import java.awt.Component
+import java.awt.event.ActionEvent
+import java.net.UnknownHostException
+import java.util.concurrent.CompletableFuture
+import javax.swing.Action
+import javax.swing.JCheckBox
+
+class SourcegraphInstanceLoginDialog(
+    project: Project?,
+    parent: Component?,
+    private val defaultInstanceUrl: String = ""
+) : DialogWrapper(project, parent, false, IdeModalityType.PROJECT) {
+
+  private var tokenAcquisitionError: ValidationInfo? = null
+  private lateinit var instanceUrlField: JBTextField
+  private lateinit var tokenField: JBTextField
+  private lateinit var customRequestHeadersField: ExtendableTextField
+  internal lateinit var codyAuthData: CodyAuthData
+
+  private val advancedAction =
+      object : DialogWrapperAction("Show advanced...") {
+        override fun doAction(e: ActionEvent) {
+          advancedSettings.isSelected = advancedSettings.isSelected.not()
+          if (advancedSettings.isSelected) {
+            setOKButtonText("Add Account")
+            putValue(NAME, "Hide advanced...")
+          } else {
+            setOKButtonText("Authorize in browser")
+            putValue(NAME, "Show advanced...")
+            tokenAcquisitionError = null
+          }
+
+          invokeLater { pack() }
+        }
+      }
+
+  private val isAcquiringToken = JCheckBox().also { it.isSelected = false }
+  private val advancedSettings = JCheckBox().also { it.isSelected = false }
+
+  init {
+    title = "Add Sourcegraph Account"
+    setOKButtonText("Authorize in browser")
+    init()
+  }
+
+  override fun createCenterPanel() = panel {
+    row {
+          cell(
+              JBLabel("Logging in, check your browser").apply {
+                icon = AnimatedIcon.Default.INSTANCE
+                foreground = getInactiveTextColor()
+              })
+        }
+        .visibleIf(isAcquiringToken.selected)
+    row("Instance URL:") {
+          textField()
+              .applyToComponent {
+                emptyText.text = "https://sourcegraph.yourcompany.com"
```

Comment:
This is an actual link! Currently it is just a parked domain, but it might [end up](https://www.bbc.com/news/technology-20627460) pointing to NSFW content sometime in the future.

Reply:
fixed

Reply:
I think it looks worse now but I guess everyone have different taste ;)
FWIW it was not clickable, so there was no risk it will redirect someone anywhere.